### ISSUE TITLE
Deployments and emulators got broken: "Error: Invalid bucket name buckets/..."

### ISSUE THREAD

**Issue by MaxGenash** [firebase/firebase-tools#6416]
### [REQUIRED] Environment info

**firebase-tools:** 10.9.2
**firebase-functions**: 3.22.0

**Platform:** Ubuntu

### [REQUIRED] Test case

1) Have a project with 368 Cloud Functions v1 of all types (onCall, onRequest, triggers, etc).
2) Execute `npx firebase deploy --debug`

### [REQUIRED] Steps to reproduce

`npx firebase deploy --debug`

### [REQUIRED] Expected behavior

The deployment works properly

### [REQUIRED] Actual behavior

Getting errors like `Invalid bucket name buckets/max-henash-joyride-personal.appspot.com` for all projects.

Previously it worked properly, but today stopped working

**Comment by MaxGenash** [comment]
Tried updating firebase-tools to 12.6.1: now emulators start, but deployment still don't work. Getting a similar problem `Got response code 400; body Failed to generate manifest from function source: Error: Invalid bucket name buckets/max-henash-joyride-personal.appspot.com`

**Comment by MaxGenash** [comment]
Based on the logs, seems like the error is caused by this request
```
9-28T19:40:23.534Z] >>> [apiv2][query] GET https://firebase.googleapis.com/v1beta1/projects/max-henash-joyride-personal/webApps/-/config [none]
[2023-09-28T19:40:24.002Z] <<< [apiv2][status] GET https://firebase.googleapis.com/v1beta1/projects/max-henash-joyride-personal/webApps/-/config 200
[2023-09-28T19:40:24.003Z] <<< [apiv2][body] GET https://firebase.googleapis.com/v1beta1/projects/max-henash-joyride-personal/webApps/-/config {"projectId":"max-henash-joyride-personal","appId":"...","databaseURL":"https://max-henash-joyride-personal-default-rtdb.firebaseio.com","storageBucket":"buckets/max-henash-joyride-personal.appspot.com","locationId":"us-central","apiKey":"...","authDomain":"max-henash-joyride-personal.firebaseapp.com","messagingSenderId":"..."}
```
https://firebase.google.com/docs/reference/firebase-management/rest/v1beta1/projects.webApps/getConfig

It returns storageBucket with slash which is not accepted later

**Comment by ottob** [comment]
I ran into this as well today and have email firebase support about it.

**Comment by maciejgoscinski** [comment]
Same issue here, entire project impossible to redeploy. Pasting the content of my support ticket:

---

I haven't made any changes to the backend of my project. Suddenly I'm unable to deploy in CI or from my local machine, due to the following error:

Error: Failed to load function definition from source: Failed to generate manifest from function source: Error: Invalid bucket name buckets/MY_BUCKET_NAME_REDACTED.appspot.com

Reverting to previous commits that used to pass the deployment also doesn't help.

I'm uploading the firebase debug log. That's the only place in the entire project where the string "buckets/MY_BUCKET_NAME_REDACTED.appspot.com" appears:

[debug] [2023-09-28T20:33:26.248Z] <<< [apiv2][body] GET https://firebase.googleapis.com/v1beta1/projects/PROJECT_NAME_REDACTED/adminSdkConfig {"projectId":"PROJECT_NAME_REDACTED","storageBucket":"buckets/MY_BUCKET_NAME_REDACTED.appspot.com","locationId":"europe-west"}

Apparently this is what is causing the build problem.
I tried the following:
- change all the storage.bucket() to actually refer to the default bucket by name
- remove all functions from index.ts entirely, leaving one empty hello world function that doesn't use anything related to Storage

**Comment by MaxGenash** [comment]
@taeold @bkendall @joehan @inlined can smb look into it plz? Seems like it's affecting many people.

Most probably it's a backend issue, but maybe you can escalate it to the responsible BE devs or implement some hotfix patch in firebase-tools (e.g. ignore the "buckets/" prefix)

**Comment by joehan** [comment]
Hey all, sorry for the inconvenience, looking into a hot fix for this ASAP.

**Comment by 50an6xy06r6n** [comment]
@joehan looks like the fix is blocked on CI failures

**Comment by joehan** [comment]
@50an6xy06r6n Yes - turns out that was not the underlying issue. We think we've identified a backend change that caused this problem, looking to roll it back ASAP

**Comment by willhlaw** [comment]
@joehan, the problem is likely to be on the backend and most likely not with firebase-tools. We did not update it today, in fact, we are on 9.23.0, and our cloud functions deploy worked yesterday, but did not this afternoon.

**Comment by 50an6xy06r6n** [comment]
I'm not sure if this is related, but when I restarted my emulator just now, it downloaded `cloud-firestore-emulator-v1.18.2.jar`, and I started seeing this error:
```
Serving at port 8902

shutdown requested via /__/quitquitquit

⬢  functions: Failed to load function definition from source: FirebaseError: Functions codebase could not be analyzed successfully. It may have a syntax or runtime error
```

**Comment by RonaldRuckus** [comment]
Same error caused by this line. No code at all:

```
exports.imageFinalized = onObjectFinalized({cpu: 2}, async (event: any) => {
});
```

**Comment by lox** [comment]
Looks like this is an upstream issue: https://github.com/firebase/firebase-tools/pull/6418#issuecomment-1740132092

Where should we be watching for future catastrophic backend breakages?

**Comment by kevintv789** [comment]
Same issue here. Spent hours trying to figure this issue out.

**Comment by jwswj** [comment]
@joehan will you update us here when the fix is rolled out? Or should we monitor somewhere else?

**Comment by ntower** [comment]
I've been experiencing this issue today, and was just able to successfully deploy a minute or so ago. Seems that they've fixed it.

**Comment by RonaldRuckus** [comment]
Can confirm that I can deploy now

**Comment by ntower** [comment]
Appears to be fixed. I've had some successes and 1 failure in the last 10 minutes. I assume the failure was a caching issue, or me hitting a server that was still getting updated.

**Comment by kevintv789** [comment]
Just tried deploying, same error. What's weird is that it allows me to deploy infrequently, only 1 out of 10 attempts work.

**Comment by KaidenR** [comment]
My local functions emulator has been failing to start up because of this error. 

I was able to get it working again by reverting back to v12.5.1 (from Aug 23) using:
`npm install --save firebase-tools@12.5.1`

It looks like v12.5.2 and all releases after it have this issue.

**Comment by lox** [comment]
From support:

> The engineering team has stated that the issue should be resolved now, can you please confirm if this is the case or if you are still experiencing issues?

They also confirmed that it was a known issue:

> Sorry about the situation regarding the deployment issues, we are aware of the situation and we are actively working on a resolution.

I've queried why it wasn't documented on the status page.

**Comment by joehan** [comment]
Hey all, this issue is resolved now. The root cause was a bad backend change, and it appears to have affected all version of firebase-tools. We've since rolled out a fix for the backend issue. If you are still having trouble deploying/emulating functions, please open a new issue and we'll investigate ASAP.

**Comment by lox** [comment]
Will there be a post-mortem made available @joehan?

**Comment by MaxGenash** [comment]
The issue has been resolved on the BE side, thanks for the quick fix

**Comment by samatcolumn** [comment]
@joehan can you explain how this affected the emulators and how it affected them across all `firebase-tools` versions? I thought that the default Admin SDK config in the emulators was constructed locally but if this affected emulators (and it did affect mine) it sounds like there's some network interaction I was not aware of?

https://github.com/firebase/firebase-tools/blob/6e9f56ea7ae7a860b8f3876f694fa429206b179d/src/emulator/adminSdkConfig.ts#L21

**Comment by samatcolumn** [comment]
Oh wait lol it was literally me:
https://github.com/firebase/firebase-tools/pull/2979

So uh ... my bad @joehan no need to explain.
